feat: initial docs for tigris sdk (#262)
tigrisdata/tigris-os-docs: docs/sdks/tigris/index.mdx

# Tigris Storage SDK

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

> Tigris Storage SDK is a simplified wrapper around Tigris Object Storage API. It
> provides a simpler interface and minimal configuration that lets you get started
> quickly and integrate Tigris into your application. It offers all the
> functionality of Tigris, but with a simpler interface.

Reviewer: ```suggestion
The Tigris Storage SDK provides a simple interface and minimal configuration
that lets you get started quickly and integrate Tigris into your application. It
is built on top of Tigris Object Storage API and offers all the functionality of Tigris.
```

## Use Cases

Tigris Storage SDK is geared (but not limited to) towards application developers
who needs to store objects that:

- Are programmatically uploaded or generated at build time, for display and
  download such as avatars, screenshots, cover images and videos
- Are larger and not practical to store in the database, such as images, videos,
  documents, etc.
- That needs to be retrieved frequently across different regions

> - break your wallet by paying egress fees

Reviewer: ```suggestion
```

> - break your wallet by paying egress fees

Author: Done

> # Tigris Storage SDK

Reviewer 2: We don't use the terminology `Tigris Data`. Tigris Data is the name of the company. We use the term `Tigris`

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

> The Tigris Storage SDK is a simplified wrapper around the Tigris Data API. It

Reviewer 2: around the `Tigris Object Storage API`

provides a simpler interface and minimal configuration that lets you get started
quickly and integrate Tigris into your application. It offers all the

> functionality of the Tigris Data, but with a simpler interface.

Reviewer 2: of Tigris

> Tigris Storage SDK is geared towards application developers who needs to store

Reviewer 2: When I read it this seems too much text. I want to immediately know how it is better. Show me what the functions look like and how they are better than AWS S3 sdk.

### Setting up your account and bucket

1. Create a Tigris account at [storage.new](https://storage.new)
2. Create a bucket at
   [console.tigris.dev/createbucket](https://console.tigris.dev/createbucket)
3. Create an access key at
   [console.tigris.dev/createaccesskey](https://console.tigris.dev/createaccesskey)

### Configure your Project

In your project root, create a `.env` file if it doesn't exist already and put
the following content in it. Replace the values with actual values your obtained
from above steps.

```bash
TIGRIS_STORAGE_ACCESS_KEY_ID=tid_access_key_id
TIGRIS_STORAGE_SECRET_ACCESS_KEY=tsec_secret_access_key
TIGRIS_STORAGE_BUCKET=bucket_name
```

### Installation

<Tabs>
<TabItem value="npm" label="NPM" default>

```bash
npm install @tigrisdata/storage
```

</TabItem>
<TabItem value="yarn" label="Yarn">

```bash
yarn add @tigrisdata/storage
```

</TabItem>
</Tabs>

And this is it, you can now use the SDK in your application. Tigris Storage SDK
supports both CommonJS and ES6 Modules syntax. That means you can use both
`require` and `import` statements to use the SDK.

<Tabs>
<TabItem value="es6mod" label="ES6 Modules" default>

```js

> import { list } from "@tigrisdata/storage";

Reviewer 2: Start with upload and show how simple it is to upload small file, automatic multipart for large files, and automatic presigned url.

const objects = await list(); // lists objects in bucket $BUCKET_NAME
console.log(objects);
```

</TabItem>
<TabItem value="commonjs" label="CommonJS Modules">

```js
const { list } = require("@tigrisdata/storage");

list().then((objects) => {
  console.log(objects);
});
```

</TabItem>
</Tabs>

> ## Examples

Reviewer 2: It is better to place the Examples section at the very end as it breaks the flow here.

# Tigris Storage SDK

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

The Tigris Storage SDK is a simplified wrapper around the Tigris Object Storage
API. It provides a simpler interface and minimal configuration that lets you get
started quickly and integrate Tigris into your application. It offers all the

> functionality of the Tigris, but with a simpler interface.

Reviewer 2: don't use `the` with Tigris

Author: Fixed

## Use Cases

Tigris Storage SDK is geared (but not limited to) towards application developers
who needs to store objects that:

- Are programmatically uploaded or generated at build time, for display and
  download such as avatars, screenshots, cover images and videos
- Are larger and not practical to store in the database, such as images, videos,
  documents, etc.
- That needs to be retrieved frequently across different regions
- break your wallet by paying egress fees

> ## Key Features

Reviewer 2: Do we need this section?

Author: Yeah I am not 100% sure about this one either, so I've removed it
